### PR TITLE
Unimodal simulation and summary figures

### PR DESCRIPTION
**Pull request by stefaneng**
# Description

  - Unimodal and multi-peak simulations
  - Plot for unimodal segment and classification

Still needs multi-peak simulations but I want to get this up for you @helen-zhu to get a chance to look at it.
Currently all of the plots are generated in `inst/unimodal_analyze_sim.R` which got pretty big. Most of the plot type themselves are placed into functions if we want to make similar plots in the same style. The results on the cluster are found here: `/hot/data/unregistered/HistogramZoo/plots/`

# Checklist

- [x] This PR does NOT contain [PHI](https://ohrpp.research.ucla.edu/hipaa/) or germline genetic data.  A repo may need to be deleted if such data is uploaded. Disclosing PHI is a [major problem](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m).
- [x] This PR does NOT contain molecular files, compressed files, output files such as images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other non-plain-text files.  To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example.
- [x] I have read the [code review guidelines](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List).
- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://confluence.mednet.ucla.edu/pages/viewpage.action?spaceKey=BOUTROSLAB&title=GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.
- [x] The name of the branch is meaningful and well formatted following the [standards](https://confluence.mednet.ucla.edu/display/BOUTROSLAB/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].
- [x] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

File: R/utils.R [hunk 1]
```diff
@@ -129,3 +129,32 @@ find_midpoint.Histogram <- function(x){
 find_midpoint.GenomicHistogram <- function(x){
   (x$consecutive_start + x$consecutive_end)/2
 }
+
+#' Loads base config file and overwrites with an optional additional file
+#'
+#' @param file optional file path to user config script
+#'
+#' @return list of yaml config
```
Comment (helen-zhu) [inline R/utils.R:137]: Is this a function that users might potentially need or is it more paper-analysis specific?

File: R/multi_peak_simulation.R [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+      peak <- peak - param
+      sd_dist <- sqrt(param) # When rate = 1
+    } else {
+      # Unif
+      param <- .sample_unif(unif_length)
+      peak <- runif(N_sim[[i]], min = -param/2, max = param/2)
+      sd_dist <- param * (1 / sqrt(12))
+    }
+
+    # To get information about peak
```
Comment (helen-zhu) [inline R/multi_peak_simulation.R:57]: Is this tentative?

File: R/multi_peak_simulation.R [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+    }
+  }
+
+  # Make histograms from peaks
+
+  # Raw un-binned peak sd
+  merged_raw_data <- unlist(sapply(multi_histogram_data, function(x) x$peak))
+  sd_raw_data <- sd(merged_raw_data)
+
+  peak_histograms <- lapply(multi_histogram_data, function(x) {
```
Comment (helen-zhu) [inline R/multi_peak_simulation.R:93]: Can potentially do `observations_to_histogram(merged_raw data)`

File: R/sim_plot_helpers.R [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+    cov.data$eps_decile <- assign.cov.factor.col(cov.data$eps_decile, c('white', covariate.col$eps.decile));
+  }
+  if ('param_decile' %in% colnames(cov.data)) {
+    cov.data$param_decile <- assign.cov.factor.col(cov.data$param_decile, c('white', covariate.col$param.decile));
+  }
+
+  if (ncol(cov.data) == 1) cov.data <- cbind(cov.data[, 1], cov.data[, 1]);
+
+  create.heatmap(
+    cov.data,
```
Comment (helen-zhu) [inline R/sim_plot_helpers.R:162]: interesting! I literally did not know you can assign colours to a matrix and create a heatmap *facepalm* - have been numerically coding my colours for years

**Comment by pboutros** [comment]
LOL so I think this is worth proposing a small example addition to create.heatmap() if you can find the time?

________________________________

UCLA HEALTH SCIENCES IMPORTANT WARNING: This email (and any attachments) is only intended for the use of the person or entity to which it is addressed, and may contain information that is privileged and confidential. You, the recipient, are obligated to maintain it in a safe, secure and confidential manner. Unauthorized redisclosure or failure to maintain confidentiality may subject you to federal and state penalties. If you are not the intended recipient, please immediately notify us by return email, and delete this message from your computer.

**Review by helen-zhu — APPROVED** [review]
Thanks so much for the pull request! Everything looks good!

**Comment by helen-zhu** [comment]
> LOL so I think this is worth proposing a small example addition to create.heatmap() if you can find the time?

I'll open an issue - if anything, I can add an example with all the time I'll save with this trick 🤣

**Comment by helen-zhu** [comment]
> > LOL so I think this is worth proposing a small example addition to create.heatmap() if you can find the time?
> 
> I'll open an issue - if anything, I can add an example with all the time I'll save with this trick 🤣

There is an example in `?create.heatmap` LOL I guess I read the plotting guide but never the help pages! Somebody should do a short talk crowdsourcing all the interesting things you can do in BPG 🍡

**Pull request merged**